accesshandler/pkg/providers/aws/sso/validate.go

```diff
@@ -84,3 +86,41 @@ func (p *Provider) ensureAccountExists(ctx context.Context, accountID string) er
 
 	return err
 }
+
+var _ providers.ConfigValidator = &Provider{}
+
+func (p *Provider) ValidateConfig() map[string]providers.ConfigValidationStep {
```

Comment:
@JoshuaWilkes I've written a new testing interface which has a bit more functionality than `gconfig.TestConfig`.

The benefits of this approach over the existing one are:
- we can specify multiple config checks, each with an ID which can be provided to Common Fate for diagnostics if a check fails
- checks can emit logs to provide more insight into what is happening during a config test
- ability to show what the checks are for a particular object without actually running them

Are you happy with this approach? If so I am keen to shift this into `gconfig` to replace TestConfig, i.e.

```diff
// Tester interface is used to run tests on config
type Tester interface {
	// TestConfig is expected to be called on a loaded config after Init has been called.
-	TestConfig(ctx context.Context) error
+	ValidateConfig() map[string]ConfigValidationStep
}

Reply:
Looks good, Checkout my other comments as one is related to testing